jwala-services/src/main/java/com/cerner/jwala/service/jvm/impl/JvmServiceImpl.java

```diff
@@ -704,7 +724,193 @@ protected void deployScriptsToUserJwalaScriptsDir(Jvm jvm, User user) throws Com
         }
 
     }
+    
+    public Jvm upgradeJDKAndDeployJvm(String jvmName, User user) {
+        boolean didSucceed = false;
+        Jvm jvm = getJvm(jvmName);
+        LOGGER.debug("Start upgradeJDKAndDeployJvm for {} by user {}", jvmName, user.getId());
+
+        historyFacadeService.write(jvm.getHostName(), jvm.getGroups(), "Starting to upgrade JDK :: JVM " +
+                jvm.getJvmName() + " on host " + jvm.getHostName(), EventType.USER_ACTION_INFO, user.getId());
+
+        //add write lock for multiple write
+        binaryDistributionLockManager.writeLock(jvmName + "-" + jvm.getId().toString());
+
+        try {
+            if (jvm.getState().isStartedState()) {

```

Comment:
Check for the JVM not in NEW state as well. We want to make sure that the JVM was already generated and deployed as a service. This method is meant to only update the JDK of an existing JVM service.

Reply:
This is fixed in the latest check-in.